src/pybind/mgr/cephadm/module.py

```diff
@@ -1980,6 +2023,18 @@ def _create_osd(self, host, cmd):
         self.cache.invalidate_host_devices(host)
         return "Created osd(s) on host '{}'".format(host)
 
+    def _calc_daemon_deps(self, daemon_type, daemon_id):
+        need = {
+            'prometheus': ['mgr', 'alertmanager', 'node-exporter'],
+            'grafana': ['prometheus'],
+            'alertmanager': ['alertmanager'],
+        }
```

Comment:
instead of a hard-coded Dict .. we might want to consider having this defined as an attribute on the spec for each `daemon_type` ?

Reply:
i considered putting this in DaemonDescription.get_dependency_types() or similar, but that's in orchestrator/_interface, and then is (somewhat) specific to cephadm at this point.  :man_shrugging: